source/sigv4_quicksort.c

```diff
@@ -0,0 +1,217 @@
+ * @param[in] pSecondItem The item to swap with @p pFirstItem.
+ * @param[in] itemSize The amount of memory per entry in the array.
+ */
+static void swap( void * pFirstItem,
+                  void * pSecondItem,
+                  size_t itemSize );
+
+/**
+ * @brief A helper function to perform quicksort on a subarray.
+ *
+ * @param[in] pArray The array to be sorted.
+ * @param[in] low The low index of the array.
+ * @param[in] high The high index of the array.
+ * @param[in] itemSize The amount of memory per entry in the array.
+ * @param[out] comparator The comparison function to determine if one item is less than another.
+ */
+static void quickSortHelper( void * pArray,
+                             size_t low,
+                             size_t high,
+                             size_t itemSize,
+                             ComparisonFunc_t comparator );
+
+/**
+ * @brief A helper function to partition a subarray using the last element
+ * of the array as the pivot. All items smaller than the pivot end up
+ * at its left while all items greater than end up at its right.
+ *
+ * @param[in] pArray The array to be sorted.
+ * @param[in] low The low index of the array.
+ * @param[in] high The high index of the array.
+ * @param[in] itemSize The amount of memory per entry in the array.
+ * @param[out] comparator The comparison function to determine if one item is less than another.
+ *
+ * @return The index of the pivot
+ */
+static size_t partition( void * pArray,
+                         size_t low,
+                         size_t high,
+                         size_t itemSize,
+                         ComparisonFunc_t comparator );
+
+/*-----------------------------------------------------------*/
+
+static void swap( void * pFirstItem,
+                  void * pSecondItem,
+                  size_t itemSize )
+{
+    uint8_t * pFirstByte = pFirstItem;
+    uint8_t * pSecondByte = pSecondItem;
+
+    if( ( pFirstItem != NULL ) && ( pSecondItem != NULL ) )
+    {
+        /* Swap one byte at a time. */
+        while( itemSize-- )
+        {
+            uint8_t tmp = *pFirstByte;
+            *pFirstByte = *pSecondByte;
+            ++pFirstByte;
+            *pSecondByte = tmp;
+            ++pSecondByte;
+        }
+    }
+}
+
+static void quickSortHelper( void * pArray,
+                             size_t low,
+                             size_t high,
+                             size_t itemSize,
+                             ComparisonFunc_t comparator )
+{
+    size_t stack[ SIGV4_WORST_CASE_SORT_STACK_SIZE ];
+    /* Low and high are first two items on the stack. */
+    size_t top = 0;
+
+    PUSH_STACK( low, stack, top );
+    PUSH_STACK( high, stack, top );
+
+    while( top > 0 )
+    {
+        size_t partitionIndex;
+        size_t lo1, lo2, hi1, hi2;
+        size_t len1, len2;
+        POP_STACK( high, stack, top );
+        POP_STACK( low, stack, top );
```

Comment:
Same here, use another variable instead of `high` and `low`, e.g. `lo` and `hi`.